creusot/tests/should_succeed/logic_functions.rs

```diff
@@ -6,16 +6,35 @@
 extern crate creusot_contracts;
 use creusot_contracts::*;
 
+logic! {
+    fn logical() -> bool { true }
+}
+
+#[ensures(logical())]
+fn main() {}
+
 mod nested {
     use creusot_contracts::*;
     logic! {
-        fn logical() -> bool { true }
+        fn nested() -> bool { true }
     }
 }
 
 logic! {
-    fn logical() -> bool { false }
+    fn arith(n: Int, b: bool) -> Int {
+        if !b {
+            -n + n - n * n
+        } else {
+            n
+        }
+    }
 }
 
-#[ensures(logical())]
-fn main() {}
+logic! {
+    fn deref_pat<'a>(o: &'a Option<Int>) -> Int {
+        match o {
+            Some(a) => *a,
```

Comment:
are you certain that this triggers the `DerefPat` branch? I really thought it was from using `&` in a pattern.